doc/modules/preprocessing.rst

```diff
@@ -559,20 +559,39 @@ parameter allows the user to specify a category for each feature to be dropped.
 This is useful to avoid co-linearity in the input matrix in some classifiers.
 Such functionality is useful, for example, when using non-regularized
 regression (:class:`LinearRegression <sklearn.linear_model.LinearRegression>`),
-since co-linearity would cause the covariance matrix to be non-invertible. 
-When this paramenter is not None, ``handle_unknown`` must be set to 
+since co-linearity would cause the covariance matrix to be non-invertible.
+When this paramenter is not None, ``handle_unknown`` must be set to
 ``error``::
 
-    >>> X = [['male', 'from US', 'uses Safari'], ['female', 'from Europe', 'uses Firefox']]
+    >>> X = [['male', 'from US', 'uses Safari'],
+    ...      ['female', 'from Europe', 'uses Firefox']]
     >>> drop_enc = preprocessing.OneHotEncoder(drop='first').fit(X)
     >>> drop_enc.categories_
     [array(['female', 'male'], dtype=object), array(['from Europe', 'from US'], dtype=object), array(['uses Firefox', 'uses Safari'], dtype=object)]
     >>> drop_enc.transform(X).toarray()
     array([[1., 1., 1.],
            [0., 0., 0.]])
 
-See :ref:`dict_feature_extraction` for categorical features that are represented
-as a dict, not as scalars.
+One might want to only drop one of the column only for feature with only 2
+categories. In this case, you can set the parameter `drop='if_binary'`.
+
+    >>> X = [['male', 'from US', 'uses Safari'],
+    ...      ['female', 'from Europe', 'uses Firefox'],
+    ...      ['female', 'from Asia', 'uses Chrome']]
```

Comment:
Maybe remove "from" and "uses" to make it more realistic. We can use pandas in documentation if we need labels, can't we?